pkg/k8shandler/configmaps.go

```diff
@@ -50,27 +68,49 @@ func createOrUpdateConfigMap(configMapName, namespace, clusterName, kibanaIndexM
 			return fmt.Errorf("Unable to get Elasticsearch cluster configMap: %v", err)
 		}
 
-		// TODO: Compare existing configMap labels, selectors and port
-		// TODO: And remember to wrap the Get/Update in a retry.RetryOnConflict
+		if configMapContentChanged(existingCM, elasticsearchCM) {
+			// Cluster settings has changed, make sure it doesnt go unnoticed
+			if err := utils.UpdateConditionWithRetry(dpl, v1alpha1.ConditionTrue, utils.UpdateUpdatingSettingsCondition); err != nil {
+				return fmt.Errorf("Unable to update Elasticsearch cluster status: %v", err)
+			}
+
+			return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				if getErr := sdk.Get(existingCM); getErr != nil {
+					logrus.Debugf("Could not get Elasticsearch %v: %v", dpl.Name, getErr)
+					return getErr
+				}
+				if updateErr := sdk.Update(elasticsearchCM); updateErr != nil {
```

Comment:
The theory of using `RetryOnConflict` is that the object returned from `Get` may have changed, and you want to preserve the existing fields in that object, and just change the fields you want to change:
```
Get(object)
object.field = value
Update(object)
```
I think if you `Update(elasticsearchCM)` you may be overwriting fields in the CM whose current state is represented by `existingCM`